packages/core/src/utils/index.js

```diff
@@ -105,4 +105,22 @@ module.exports = {
 
     return text.join('').trim();
   },
+
+  /**
+   * Applies an asynchronous function for each element in an array.
+   * Each application evaluation is done sequentially. That is, an asynchronous function
+   * application of an element is evaluated only when the previous applications
+   * have finished.
+   *
+   * @param {Array} array The array to be iterated over
+   * @param func The asynchronous function to be applied to each element
+   * @returns {Promise<void>} A Promise that resolves once every application has been evaluated
+   */
+  async sequentialAsyncForEach(array, func) {
+    for (let i = 0; i < array.length; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
+      await func(array[i]);
+      // eslint-enable-next-line no-await-in-loop
```

Comment:
👀

Reply:
Yeah, I think this is kind of unavoidable. We need to await each async function to finish before moving onwards to the next element in order for the async functions to execute sequentially (use case: sequential generation). Had to make do with disabling that eslint rule.

Reference to this approach:
- https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop
- https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404

Reply:
referring to the `eslint-enable-next-line` in particular

Reply:
Oh god haha my mistake 😅 TIL I don't need to do that